rnd/autogpt_server/autogpt_server/server/server.py

```diff
@@ -373,7 +475,59 @@ async def create_new_graph(cls, graph: Graph) -> Graph:
             link.source_id = id_map[link.source_id]
             link.sink_id = id_map[link.sink_id]
 
-        return await create_graph(graph)
+        return await autogpt_server.data.graph.create_graph(graph)
+
+    @classmethod
+    async def update_graph(
+        cls, graph_id: str, graph: autogpt_server.data.graph.Graph
+    ) -> autogpt_server.data.graph.Graph:
+        # Sanity check
+        if graph.id and graph.id != graph_id:
+            raise HTTPException(400, detail="Graph ID does not match ID in URI")
+
+        # Determine new version
+        existing_versions = await autogpt_server.data.graph.get_graph_all_versions(
+            graph_id
+        )
+        if not existing_versions:
+            raise HTTPException(400, detail=f"Unknown graph ID '{graph_id}'")
+        graph.version = max(g.version for g in existing_versions) + 1
+
+        if not graph.is_template:
+            graph.is_active = True
+        else:
+            graph.is_active = False
```

Comment:
```suggestion
        graph.is_active = not graph.is_template
```

Reply:
I think this logic may be flawed: what if a user decides to revert their template to an earlier version. How would we execute that? Do we need to start using `is_active` the same way as for graphs, so that the active template version has `is_active = True`?